src/backends/fvm_multicore.hpp

```diff
@@ -50,30 +50,38 @@ struct backend {
         // the invariant part of the matrix diagonal
         array invariant_d;         // [μS]
 
+        const_view solution;
+
         std::size_t size() const { return p.size(); }
 
         matrix_state() = default;
 
-        matrix_state(const_iview p, const_iview cell_index):
-            p(p), cell_index(cell_index),
-            d(size()), u(size()), rhs(size())
-        {}
-
-        matrix_state(const_iview p, const_iview cell_index, array cap, array cond):
-            p(p), cell_index(cell_index),
-            d(size()), u(size()), rhs(size()),
-            cv_capacitance(std::move(cap)),
-            face_conductance(std::move(cond))
+        //matrix_state(const_iview p, const_iview cell_index, array cap, array cond):
+        matrix_state( const std::vector<size_type>& p,
+                      const std::vector<size_type>& cell_idx,
+                      const std::vector<value_type>& cap,
+                      const std::vector<value_type>& cond):
+            p(memory::make_const_view(p)),
+            cell_index(memory::make_const_view(cell_idx)),
+            d(size(), 0), u(size(), 0), rhs(size()),
+            cv_capacitance(memory::make_const_view(cap)),
+            face_conductance(memory::make_const_view(cond))
         {
-            auto n = d.size();
-            invariant_d = array(n, 0);
-            for (auto i: util::make_span(1u, n)) {
-                auto gij = face_conductance[i];
-
-                u[i] = -gij;
-                invariant_d[i] += gij;
-                invariant_d[p[i]] += gij;
+            if (has_fvm_state()) {
```

Comment:
I removed the `has_fvm_state` completely, and enforce that capacitance and conductance are provided for each compartment/face.
Ditto for the gpu implementations, and the changes this requires in the unit tests.